src/librustc_front/hir.rs

```diff
@@ -50,7 +50,59 @@ use print::pprust;
 use util;
 
 use std::fmt;
-use serialize::{Encodable, Encoder, Decoder};
+use std::hash::{Hash, Hasher};
+use serialize::{Encodable, Decodable, Encoder, Decoder};
+
+/// Identifier in HIR
+#[derive(Clone, Copy, Eq)]
+pub struct Ident {
+    /// Hygienic name (renamed), should be used by default
+    pub name: Name,
+    /// Unhygienic name (original, not renamed), needed in few places in name resolution
+    pub unhygienic_name: Name,
+}
+
+impl Ident {
+    pub fn from_name(name: Name) -> Ident {
```

Comment:
`from_hygienic_name`? Since the caller is guaranteeing that the input is already hygienic? Otherwise just a comment explaining the danger would be good.